src/components/ActivityForm.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+
+export default class ActivityForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      classNo: props.activity ? props.activity.classNo : '',
+      day: props.activity ? props.activity.day : '',
+      name: props.activity ? props.activity.name : '',
+      room: props.activity ? props.activity.room : '',
+      teacher: props.activity ? props.activity.teacher : '',
+      error: ''
+    };
+  };
+  onNameChange = (e) => {
+    const name = e.target.value;
+    this.setState(() => ({name}));
+  };
+  onTeacherChange = (e) => {
+    const teacher = e.target.value;
+    this.setState(() => ({teacher}));
+  };
+  onRoomChange = (e) => {
+    const room = e.target.value;
+    this.setState(() => ({room}));
+  };
+  onClassNoChange = (e) => {
+    const classNo = e.target.value;
+    if (!classNo || classNo.match(/^[1-9]\d*$/)) {
+      this.setState(() => ({classNo}));
+    };
+  };
+  onDayChange = (e) => {
+    const day = e.target.value;
+    this.setState(() => ({day}));
+  };
+  onSubmit = (e) => {
+    e.preventDefault();
+    if (!this.state.name || !this.state.day || !this.state.classNo) {
+      this.setState(() => ({error: 'Please provide Activity Name, Day and Class Number!'}))
```

Comment:
the same withh passing here an object. The ule is, if you are not rely on previous state, pas just an object.